cmr/cmr-tool.cwl

```diff
@@ -2,50 +2,39 @@ cwlVersion: v1.2
-          '''
-          f.write(stac_output)
-
+  InlineJavascriptRequirement: {}
+  DockerRequirement:
+    dockerPull: ghcr.io/unity-sds/unity-data-services:3.3.1
+  EnvVarRequirement:
+    envDef:
+      GRANULES_SEARCH_DOMAIN: 'CMR'
+      CMR_BASE_URL: 'https://cmr.earthdata.nasa.gov'
+      COLLECTION_ID: $(inputs.cmr_collection)
+      LIMITS: $(inputs.limits || '-1')
+      DATE_FROM: $(inputs.cmr_start_time)
+      DATE_TO: $(inputs.cmr_stop_time)
+      FILTER_ONLY_ASSETS: 'TRUE'
 
 inputs:
   cmr_collection:
     type: string
-    inputBinding:
-      position: 1
-      prefix: --collection
   cmr_start_time:
     type: string
-    inputBinding:
-      position: 2
-      prefix: --start-time
   cmr_stop_time:
     type: string
-    inputBinding:
-      position: 3
-      prefix: --end-time
   cmr_edl_user:
     type: string?
-    inputBinding:
-      position: 4
-      prefix: --edl-user
   cmr_edl_pass:
     type: string?
-    inputBinding:
-      position: 5
-      prefix: --edl-pass
+  limits:
+    type: string?
 
 #stac json catalog are the outputs
 outputs:
   cmr_results:
-      outputBinding:
-        glob: "cmr-results.json"
-      type: File
+    type: stdout
```

Comment:
Will this work? I'm not sure how the CMR output will get to the next step but i'm not a master of CWL either... maybe this "just works" - have we tested this on a  HUGE list of files?

Reply:
The stdout gets saved to cmr-results.json https://github.com/unity-sds/sounder-sips-chirp-workflows/pull/3/files#diff-c4c896ffe11f5dff538c92b6f688a1728616a9f8fd688cba58497bb7c52f5520R7

Works for 16 day time range

cmr_start_time: "2021-02-01T00:00:00Z"
cmr_stop_time: "2021-02-16T00:00:00Z"

Reply:
oh interesting, didn't realize that stdout could be a file